### DISCUSSION TITLE
Generate latest loco binary using latest loco_cli lib from master branch ?

### DISCUSSION THREAD

**Discussion by tjyang** [loco-rs/loco#240] [Q&A] [answered]
-  Is following steps correct ? 

```
git pull
cargo build --release
cd loco-cli
cargo install --path .  
```

- I can't see relationship from loco-cli/Cargo.tom using loco_lib library framework just got compiled.

```
[me@rocky8t01d loco]$ ls -ld /home/me/.cargo/bin/loco*
-rwxrwxr-x 1 me me 6229456 Dec 26 10:41 /home/me/.cargo/bin/loco         #  This the one compiled above.
-rwxrwxr-x 1 me me 6233552 Dec 26 10:33 /home/me/.cargo/bin/loco.0.2.5 # this is from crates.io install.
[me@rocky8t01d loco]$
```

**Comment by kaplanelad** [comment]
correct for what?
what are you trying to achieve?

**Reply by tjyang** [reply]
Goal: Trying to learn loco/Rust by modifying  source code inside loco/src an loco/loco-cli/src

```
[me@rocky8t01d loco]$ ls  src loco-cli/src
loco-cli/src:
bin  generate.rs  git.rs  lib.rs  prompt.rs  snapshots

src:
app.rs     boot.rs  config.rs   doctor.rs       gen      logger.rs  prelude.rs  snapshots  testing.rs
auth       cli.rs   controller  environment.rs  hash.rs  mailer     redis.rs    task.rs    validation.rs
banner.rs  concern  db.rs       errors.rs       lib.rs   model.rs   schema.rs   tera.rs    worker.rs
[me@rocky8t01d loco]$ pwd
/home/me/github/learnrust/loco
[me@rocky8t01d loco]$
```

**Comment by kaplanelad** [comment, marked as answer]
loco framework and loco cli not sharing libs. basically they can live in a different repo

**Reply by tjyang** [reply]
@kaplanelad , Thanks for the clarification.

I was assuming loco-cli will link  to loco-rs library/framework. only loco-cli(name as loco binary) generated .Cargotoml file  will link to loco-rs lib/framework. I can use blo-cli example(src=example/demo/*) to achieve my goal.

- blo-cli example are linking back  to loco-rs two directory level up.
```
[me@rocky9t01a loco]$ grep loco-rs examples/demo/Cargo.toml
loco-rs = { version = "*", path = "../../" }
loco-rs = { version = "*", path = "../../", features = ["testing"] }
[me@rocky9t01a loco]$
'''

**Comment by kaplanelad** [comment]
@tjyang I think you are confusing between the CLI of the framwork you can see it [here](https://github.com/loco-rs/loco/blob/master/src/cli.rs)

and the starter CLI when you run `loco new`
